amaranth/lib/wiring.py

```diff
@@ -1574,6 +1576,12 @@ def connect_dimensions(dimensions, *, out_path, in_path):
                                        out_path=(*out_path, index), in_path=(*in_path, index))
             assert out_member.dimensions == in_member.dimensions
             connect_dimensions(out_member.dimensions, out_path=out_path, in_path=in_path)
+
+    # If no connections were made, and there were no outputs actually found, issue a diagnostic
+    # as this is likely in error.
+    if len(connections) == 0 and not any_out:
```

Comment:
This does fire on an empty signature, right? I think connecting any number of empty signatures is important to allow, for the same reason we allow zero-bit signals; it is useful in generic code to stub out unused interfaces.

Reply:
It did indeed. We now track `any_in` as well.